experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/SparkListener.java

```diff
@@ -18,79 +22,68 @@
 import org.apache.spark.scheduler.SparkListenerJobStart;
 import org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionEnd;
 import org.apache.spark.sql.execution.ui.SparkListenerSQLExecutionStart;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import marquez.client.Backend;
-import marquez.spark.agent.lifecycle.ExecutionContext;
-import marquez.spark.agent.lifecycle.RddExecutionContext;
-import marquez.spark.agent.lifecycle.SparkSQLExecutionContext;
-import marquez.spark.agent.transformers.ActiveJobTransformer;
-import marquez.spark.agent.transformers.PairRDDFunctionsTransformer;
-
 
+@Slf4j
 public class SparkListener {
-  private static final Logger logger = LoggerFactory.getLogger(SparkListener.class);
-
-  private static final Map<Long, SparkSQLExecutionContext> sparkSqlExecutionRegistry = Collections.synchronizedMap(new HashMap<>());
-
-  public static SparkSQLExecutionContext getSparkSQLExecutionContext(long executionId) {
-    SparkSQLExecutionContext sparkSQLExecutionContext = sparkSqlExecutionRegistry.get(executionId);
-    if (sparkSQLExecutionContext == null) {
-      sparkSQLExecutionContext = new SparkSQLExecutionContext(executionId, marquezContext);
-      sparkSqlExecutionRegistry.put(executionId, sparkSQLExecutionContext);
-    }
-    return sparkSQLExecutionContext;
-  }
-
-  private static final Map<Integer, ExecutionContext> rddExecutionRegistry = Collections.synchronizedMap(new HashMap<>());
-
-  public static ExecutionContext getExecutionContext(int jobId) {
-    ExecutionContext rddExecutionContext = rddExecutionRegistry.get(jobId);
-    if (rddExecutionContext == null) {
-      rddExecutionContext = new RddExecutionContext(jobId, marquezContext);
-      rddExecutionRegistry.put(jobId, rddExecutionContext);
-    }
-    return rddExecutionContext;
-  }
-
-  public static ExecutionContext getExecutionContext(int jobId, long executionId) {
-    ExecutionContext executionContext = getSparkSQLExecutionContext(executionId);
-    rddExecutionRegistry.put(jobId, executionContext);
-    return executionContext;
-  }
-
-  private static Map<RDD<?>, Configuration> outputs = Collections.synchronizedMap(new HashMap<>());
-
-  public static Configuration removeConfigForRDD(RDD<?> rdd) {
-    return outputs.remove(rdd);
-  }
-
+  private static final Map<Long, SparkSQLExecutionContext> sparkSqlExecutionRegistry =
+      Collections.synchronizedMap(new HashMap<>());
+  private static final Map<Integer, ExecutionContext> rddExecutionRegistry =
+      Collections.synchronizedMap(new HashMap<>());
+  private static WeakHashMap<RDD<?>, Configuration> outputs = new WeakHashMap<>();
```

Comment:
I think a weakmap is okay here but not fully sure

Reply:
As long as the value (`Configuration`) does not have a reference to the key (`RDD`) I think we are good.
I don't think this can be the case here.